src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml

```diff
@@ -2289,6 +2337,96 @@ components:
+      allOf:
+        - $ref: "#/components/schemas/ResultSetCursorMixin"
+        - type: object
+          required:
+            - items
+          properties:
+            items:
+              type: array
+              items:
+                $ref: "#/components/schemas/AccountResourcePreferencesResponseItem"
+
+    AccountResourcePreferencesResponseItem:
+      type: object
+      required:
+        - resource_address
+        - resource_preference_rule
+        - last_updated_at_state_version
+      properties:
+        resource_address:
+          $ref: "#/components/schemas/Address"
+        resource_preference_rule:
+          type: string
+          enum:
+          - Allowed
+          - Disallowed
+        last_updated_at_state_version:
+          $ref: "#/components/schemas/LastUpdatedAtStateVersion"
+
+    AccountAuthorizedDepositorsCollection:
+      description: Account resource preferences collection
+      allOf:
+        - $ref: "#/components/schemas/ResultSetCursorMixin"
+        - type: object
+          required:
+            - items
+          properties:
+            items:
+              type: array
+              items:
+                $ref: "#/components/schemas/AccountAuthorizedDepositorsResponseItem"
+
+    AccountAuthorizedDepositorType:
+      type: string
+      enum:
+        - ResourceBadge
+        - NonFungibleResourceBadge
+
+    AccountAuthorizedDepositorsResponseItem:
+      type: object
+      required:
+        - badge_type
+      properties:
+        badge_type:
+          $ref: "#/components/schemas/AccountAuthorizedDepositorType"
+      discriminator:
+        propertyName: badge_type
+        mapping:
+          # Must match AccountAuthorizedDepositorType
+          ResourceBadge: "#/components/schemas/AccountAuthorizedDepositorsResourceBadge"
+          NonFungibleResourceBadge: "#/components/schemas/AccountAuthorizedDepositorsNonFungibleResourceBadge"
+    AccountAuthorizedDepositorsResourceBadge:
+      allOf:
+        - $ref: "#/components/schemas/AccountAuthorizedDepositorsResponseItem"
+        - type: object
+          required:
+            - resource_address
+            - last_updated_at_state_version
+          properties:
+            resource_address:
+              $ref: "#/components/schemas/Address"
+            last_updated_at_state_version:
+              $ref: "#/components/schemas/LastUpdatedAtStateVersion"
+    AccountAuthorizedDepositorsNonFungibleResourceBadge:
+      allOf:
+        - $ref: "#/components/schemas/AccountAuthorizedDepositorsResponseItem"
+        - type: object
+          required:
+            - resource_address
+            - non_fungible_id
+            - last_updated_at_state_version
+          properties:
+            resource_address:
+              $ref: "#/components/schemas/Address"
+            non_fungible_id:
+              $ref: "#/components/schemas/NonFungibleId"
+            last_updated_at_state_version:
```

Comment:
Shouldn't `resource_address` and `last_updated_at_state_version` be defined in terms of `AccountAuthorizedDepositorsResponseItem`?

Reply:
Tried to do so... but that resulted in some strange errors when generating models.. after moving them to inherited objects it worked fine.

Reply:
Damn OAS generator... OK, let's leave it as it is.

Reply:
Give me a shout if you'd like to pair on fixing the OAS error

Reply:
Although actually on reflection, I think it's fine for hte resource address to be separate under each option, as conceptually the key is either a (resource address) or a (global non fungible id) and the fact they share a resource address is coincidental.

Perhaps though I'd like to air a suggestion to us separating the entry into three parts:
* key: AccountAuthorizedDepositor
* value: { is_authorized: true }
* last_updated_at_state_version

This makes the `key` / `value` separation more obvious, and allows us more flexibility in future.

But happy for us to stick with what we have for consistency.